### DISCUSSION TITLE
Help Request: Library Import Issues in Teammates Application Contribution

### DISCUSSION THREAD

**Discussion by Ravinkumar2264** [TEAMMATES/teammates#12674] [Help requests]
### Environment

- **Commit SHA:** 
- **Operating System:Windows**
- **JDK Version:17**
- **Node.js Version:**
- **Firefox / Chrome Version (if applicable):**

### Intention

I am trying to contribute to the teammates application. I have followed all the steps given in the documentation https://github.com/TEAMMATES/teammates/blob/master/docs/development.md , but I can't debug the code, and the library files are not imported properly. If I need to set any paths to debug this application, I am using the STS IDE. Can anyone help me fix this library issues?

### Resolution Attempt

![Screenshot (31)](https://github.com/TEAMMATES/teammates/assets/153491308/b6c5e29c-a132-4f8c-8fe9-89345d071b85)

**Comment by weiquu** [comment]
Hi @Ravinkumar2264, have you also followed the steps in the [Setting Up](https://teammates.github.io/teammates/setting-up.html) page? Can I also check if the dev server runs, and if not, what is the error?

**Reply by Ravinkumar2264** [reply]
The Dev Server is running properly, but I can't debug the code, and the library files are not imported correctly.

**Reply by weiquu** [reply]
Hi @Ravinkumar2264, if the website is running properly (i.e. you are able to complete actions without errors), then it's likely that the issue lies with your IDE. I'm using VSCode, so I run the "Java: Clean Java Language Server Workspace" command when I encounter this - you might want to search online for the command specific to your IDE.

If however the website is not running properly for you, please provide the error shown (and do ensure you've followed the steps both in the Setting Up page as well as the Development page).

**Reply by wkurniawan07** [reply]
@Ravinkumar2264 I see that you're using an IDE called Spring Tool Suite. If the name implies what I think it is, there may be problems because the backend is not a Spring Boot or even Spring Framework backend.

**Reply by Ravinkumar2264** [reply]
![image](https://github.com/TEAMMATES/teammates/assets/153491308/d0e3bea6-e95f-4464-9f17-a16ee6a1aafe)
@wkurniawan07 @weiquu  Now I can run both the front-end server and backend servers, but I can't debug the code. When I debug this code, I am encountering these errors. Could you please help me debug the Java code?

**Reply by weiquu** [reply]
Hmm I don't have experience setting breakpoints while running the dev server, usually the logs from there (i.e. the process running `./gradlew serverRun`) gives enough info to debug. You should be able to see very comprehensive details about the data being passed through as well as any errors that occur.

Let me know if you don't see that info while using the website

**Reply by wkurniawan07** [reply]
@Ravinkumar2264 debugging code is not a skill specific to this system, so you can find plenty of explanations / tutorials online. I will just say that in the screenshot that you shared, there are no indication of errors, those are logs normally seen when the backend server just booted up.

**Comment by Ravinkumar2264** [comment]
@weiquu @wkurniawan07 Could you provide the architecture diagram for this project if it is available?

**Reply by domlimm** [reply]
Not sure if this is what you're looking for - https://teammates.github.io/teammates/design.html. You may find most things that you will need to start contributing.
